src/main/java/de/codingair/tradesystem/trade/Trade.java

```diff
@@ -294,31 +302,33 @@ private void finish() {
                 guis[0].setItem(slot, null);
 
                 if(i0 != null && !i0.getType().equals(Material.AIR)) {
-                    int rest = fit(players[0], i0);
+                    int rest = fit(player1, i0);
 
                     if(rest <= 0) {
-                        players[0].getInventory().addItem(i0);
+                        player1.getInventory().addItem(i0);
                     } else {
                         ItemStack toDrop = new ItemBuilder(i0).setAmount(rest).getItem();
                         i0.setAmount(i0.getAmount() - rest);
 
-                        if(i0.getAmount() > 0) players[0].getInventory().addItem(i0);
-                        Environment.dropItem(toDrop, players[0]);
+                        if(i0.getAmount() > 0) player1.getInventory().addItem(i0);
+                        Environment.dropItem(toDrop, player1);
                     }
+                    getTradeLog().log(player1, player2, player1.getName() + " received item " + i0.getAmount() + " "+ i0.getType());
```

Comment:
Synchron SQL call -> Might impact the server performance if the database is the bottleneck.
-> Asynchron call with a callback

Reply:
done